search_service/proxy/atlas.py

```diff
@@ -45,71 +35,187 @@ def __init__(self, *,
 
-            table_results.append(table)
+        :param query_params: A dictionary of query parameters needed to be pass to Basic Search Post method of Atlas
+        :return: List of Table objects and approximate count of entities matching in Atlas
+        """
 
-        return table_results
+        try:
+            # Fetch the table entities based on query terms
+            search_results = self.atlas.search_basic.create(data=query_params)
+        except BadRequest as ex:
+            LOGGER.error(f"Fetching Tables Failed : {str(ex)}")
+            return [], 0
+
+        if not len(search_results.entities):
+            return [], 0
+
+        # noinspection PyProtectedMember
+        results_count = search_results._data.get("approximateCount")
+
+        results = self._prepare_tables(search_results.entities, enhance_metadata=False)
+
+        return results, results_count
+
+    def _prepare_basic_search_query(self, limit: int, page_index: int, query_term: Optional[str] = None,
+                                    filters: Optional[List[Tuple[str, str, str]]] = None,
+                                    operator: Optional[str] = None,
+                                    classifications: Optional[List[str]] = None,
+                                    entity_type: str = None) -> Dict[str, Any]:
+        """
+        Render a query for Atlas Basic Search API.
+
+        :param query_term: Search Query Term
+        :param limit:
+        :param page_index:
+        :param fitlers: Optional list of tuples (field, condition, value) that will translate into entityFilters for
+        narrowing down search results
+        :param operator:
+        :param entity_type: Which kind of entity this query will look for
+        :return: Dictionary object prepared for Atlas client basic_search
+        """
+        if not entity_type:
+            entity_type = self.ATLAS_TABLE_ENTITY
+
+        query: Dict[str, Any] = dict(typeName=entity_type,
+                                     excludeDeletedEntities=True,
+                                     limit=limit,
+                                     offset=page_index * self.page_size,
+                                     sortBy='popularityScore',
+                                     sortOrder='DESCENDING',
+                                     attributes=['description', 'comment', 'popularityScore'])
+
+        if query_term:
+            query_term = f'*{query_term}*'
+            query_term = sub('\\*+', '*', query_term)
+
+            query['query'] = query_term
+
+        # filters and query_term shouldn't be mixed
+        if filters and not query_term:
+            condition = operator or 'AND'
+            criterion: List[Dict[str, str]] = list()
+
+            for _query_filter in filters:
+                attribute_name, operator, attribute_value = _query_filter
+
+                # filters perform much better when wildcard is dot, not star
```

Comment:
Really great catch.

Btw, does this query rewriting have any user-facing impact?

Reply:
because exact matches are more costly (performance-wise) I've just assumed that user types name of table (but can be a fragment of name) and either way we make a `contains` query on that